scandium-core/src/main/java/org/eclipse/californium/scandium/dtls/Record.java

```diff
@@ -839,21 +865,17 @@ public void setSequenceNumber(long sequenceNumber) throws GeneralSecurityExcepti
 	 * @return the fragment length excluding record headers
 	 */
 	public int getFragmentLength() {
-		return fragmentLength;
+		final byte[] bytes = fragmentBytes;
+		return bytes == null ? 0 : bytes.length;
 	}
 
 	/**
 	 * Set associated session.
 	 * 
-	 * Take over peer address from session.
-	 * 
 	 * @param session session to be set
 	 */
 	public void setSession(DTLSSession session) {
 		this.session = session;
-		if (session != null && this.peerAddress == null) {
-			this.peerAddress = session.getPeer();
-		}
```

Comment:
This was something which has been added recently, and finally this does not make sense, right ?

Reply:
```
public synchronized void setSession(DTLSSession session) {
	this.session = session;
	if (session != null) {
		this.peerAddress = null;
	}
}

public InetSocketAddress getPeerAddress() {
	if (session != null) {
		return session.getPeer();
	} else if (peerAddress != null) {
		return peerAddress;
	} else {
		throw new IllegalStateException("Record does not have a peer address");
	}
}
```

That was the original implementation. So this cleanup PR is more a "next step" then a "reverse step".